src/common/exif.cc

```diff
@@ -1717,8 +1717,7 @@ int dt_exif_get_thumbnail(const char *path, uint8_t **buffer, size_t *size, char
   }
   catch(Exiv2::AnyError &e)
   {
-    std::string s(e.what());
-    std::cerr << "[exiv2 dt_exif_get_thumbnail] " << path << ": " << s << std::endl;
+    dt_print(DT_DEBUG_IMAGEIO, "[exiv2 dt_exif_get_thumbnail] %s: %s\n", path, e.what());
```

Comment:
Those are plain errors so I bet it is preferable to have them always displayed, no?

Reply:
This is exactly what I wrote about in a comment to PR, the lack of Exif data and a thumbnail is far from always an error. darktable is not limited to processing only images created by digital devices, it supports many other formats (and even JPEG can be exported from a drawing program and do not contain Exif data).

Printing messages to stderr/logfile without debug print enabled should be reserved only for really critical messages about what is always an error and a fairly serious error.

Reply:
Yes I understand and if I agree for thumbnails it seems strange for other Exif data. To me Exif are quite critical for dt proper behavior. So the second instance in the diff seems strange to me, I didn't look at the whole file and maybe I'm missing the fact that the exception is for some non critical Exif data.

Reply:
Well, dt supports many image formats where Exif data is simply not supported by the format. I myself sometimes edited such images using dt and I did not notice any improper behavior. Yes, certain modules that rely on metadata from the camera to function properly will not be able to work with such images. This is understandable and expected. But in everything else dt perfectly processes images without Exif. In such cases, constant complaints to stderr simply clog the output with messages that in this situation do not report valid errors.

Reply:
Thanks for the clarifications.

Reply:
Just curious.  Do the "modules that rely on metadata from the camera to function properly" print messages and/or provide an indication in the UI when they encounter images that don't contain the required metadata?

Reply:
@pcanning Yes. This is, for example, lens correction. This module simply will not work if it does not receive the necessary information about the camera and lens (or built-in correction parameters). And yes, when you try to enable this module, you will receive a warning about the lack of necessary data.